### PR TITLE
Implement dropdown menu in Toolbar and various UI improvements

### PR DESCRIPTION
**Pull request by joey-PHC-dev**
- Added DropdownMenu component to the Toolbar.
- Implemented a profile dropdown menu with links to Wallet, Inventory, Tickets, and Settings.
- Styled the dropdown menu to match the provided design, including icons and hover effects.
- Fixed alignment and spacing issues in the Toolbar.
- Added hover effect for the profile icon in the Toolbar.
- Adjusted the spacing between the Profile Name and the View Profile link in the dropdown menu.
- Included temporary page reload workaround for navigation on Notification and Message icons click in the Toolbar.
- Improved the positioning and styling of the dropdown menu to ensure it is displayed correctly below the profile icon.

### CODE REVIEW

File: frontend/src/components/Profile/UserProfileExternal.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+const UserProfileExternal: React.FC<UserProfileExternalProps> = ({ userId }) => {
+    const [user, setUser] = useState<User | null>(null);
+
+    useEffect(() => {
+        let unsubscribe: (() => void) | null | undefined = null;
+
+        const setupSubscription = async () => {
+            unsubscribe = await subscribeToUserModelSubject((user) => {
+                if (user.id === userId) {
```
Comment (nick-cd) [inline frontend/src/components/Profile/UserProfileExternal.tsx:52]: If I understand the logic correctly, I think we should raise this condition to the parent component. This way, we can make sure the external profile component _only_ renders when ID is some other co-quest user.

File: frontend/src/components/Profile/UserProfileInternal.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+const UserProfileInternal: React.FC = () => {
+    const [user, setUser] = useState<User | null>(null);
+    const [menuVisible, setMenuVisible] = useState(false);
+    const [menuPosition, setMenuPosition] = useState({ top: 0, left: 0 });
+    const menuIconRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        let unsubscribe: (() => void) | null | undefined = null;
+
+        const setupSubscription = async () => {
```
Comment (nick-cd) [inline frontend/src/components/Profile/UserProfileInternal.tsx:146]: I also suggest raising the current user subscription to the parent component and pass the user down as a prop. It'll clean up the child components.

**Review by nick-cd — CHANGES_REQUESTED** [review]

File: frontend/src/components/Profile/UserProfileExternal.tsx (hunk 1 above)
Comment (joey-PHC-dev) [reply frontend/src/components/Profile/UserProfileExternal.tsx]: Moved user data subscription logic from UserProfileInternal and UserProfileExternal to UserProfile component

File: frontend/src/components/Profile/UserProfileInternal.tsx (hunk 2 above)
Comment (joey-PHC-dev) [reply frontend/src/components/Profile/UserProfileInternal.tsx]: Updated UserProfileInternal to receive user data as a prop and refactored the component accordingly

**Pull request merged**